server/src/processing/HeaderFooterDetectionModule/HeaderFooterDetectionModule.ts

```diff
@@ -45,60 +41,102 @@ export class HeaderFooterDetectionModule extends Module<Options> {
+          const v: number[] = page.verticalOccupancy.map(this.boolToInt);
+          occupancyAcrossWidth = utils.addVectors(occupancyAcrossWidth, v);
+        });
+        
+        // UNCOMMENT THESE TO EXPORT OCCUPANCIES INTO EXTERNAL CSV FILES
+        // writeFileSync("horizontal.csv", occupancyAcrossWidth.join(";"), {encoding: 'utf-8'})
+        // writeFileSync("vertical.csv", occupancyAcrossHeight.join(";"), {encoding: 'utf-8'})
+        this.setMargins(doc, occupancyAcrossHeight, occupancyAcrossWidth);
+        groupOfPage.forEach(page => {
+          this.setHeaderAndFooter(doc, page);
+          const headerElements: Element[] = page.getElementsSubset(
+            new BoundingBox(0, 0, page.width, doc.margins.top),
+          );
+          const footerElements: Element[] = page.getElementsSubset(
+            new BoundingBox(0, doc.margins.bottom, page.width, page.height - doc.margins.bottom),
+          );
+          for (const element of footerElements) {
+            element.properties.isFooter = true;
+          }
+          for (const element of headerElements) {
+            element.properties.isHeader = true;
+          }
+        });
+        occupancyAcrossHeight = [];
+        occupancyAcrossWidth = [];
+      });
+    logger.debug('Done with marginals detection.');
+    return doc;
+  }
+
+  private isComputable(doc: Document): boolean {
     const alreadyExist: boolean =
       doc.pages
         .map(p => {
           return p.elements.filter(e => e.properties.isHeader || e.properties.isFooter).length;
         })
         .reduce((a, b) => a + b, 0) > 0;
 
-    if (doc.pages.length === 1) {
+    let nbPageToTreat: number = this.countPageToTreat(doc.pages.length);
+    if (nbPageToTreat <= 1) {
       logger.warn(
         'Not computing marginals (headers and footers)' +
-          'the document only has 1 page (not enough data).',
+          'the document has only 1 page to check (not enough data).',
       );
-      return doc;
+      return false;
     } else if (this.options.maxMarginPercentage === undefined) {
       logger.info(
         'Not computing marginals (headers and footers); maxMarginPercentage setting not found in the configuration.',
       );
-      return doc;
+      return false;
     } else if (alreadyExist) {
       logger.warn(
         'Not computing marginals (headers and footers): header and footer data already exists.',
       );
-      return doc;
+      return false;
     }
     logger.info(
       'Detecting marginals (headers and footers) with maxMarginPercentage:',
       this.options.maxMarginPercentage,
       '...',
     );
-
-    let occupancyAcrossHeight: number[] = [];
-    let occupancyAcrossWidth: number[] = [];
-
-    function boolToInt(p) {
-      if (p) {
-        return 1;
-      } else {
-        return 0;
-      }
+    return true;
+  }
+  private boolToInt(p) {
+    if (p) {
+      return 1;
+    } else {
+      return 0;
```

Comment:
1. Missing param type & return type
2. p is too short param name...  private boolToInt(value: boolean) : boolean
2. One line improvement -->  return p ? 1 : 0

Reply:
Ok thanks, I'm gonna make the change!